README.md

```diff
@@ -21,30 +37,119 @@ from fastapi_nextauth_jwt import NextAuthJWT
+
+## NextAuth.js v4 Compatibility
+
+For NextAuth.js v4 applications, use the `NextAuthJWTv4` class:
+
+```python
+from fastapi_nextauth_jwt import NextAuthJWTv4
+
+JWT = NextAuthJWTv4(
+    secret=os.getenv("NEXTAUTH_SECRET")
+)
 ```
 
-There are a few configuration options available in the NextAuthJWT constructor, but the most important one is `secret`,
-which should be equivalent to `NEXTAUTH_SECRET` on the Next.js side. Please don't hardcode the secret like this in a real application,
-but rely on the `NEXTAUTH_SECRET` env var :)
+This provides compatibility with the v4 token format and default settings, including:
+- Uses `"A256GCM"` encryption algorithm
+- Sets specific key derivation parameters
+- Uses v4-compatible cookie names
+- Disables salt auto-appending
+
+## Security Best Practices
 
-It is also possible to enable or disable CSRF protection using `csrf_prevention_enabled`. 
-If this is not set, this will looks at the ENV environment variable. If this is `dev` then CSRF protection will be disabled.
-It is also possible to customize the HTTP verbs to which CSRF protection is applied.
+1. **Environment Variables**: Always use environment variables for sensitive values:
+   ```python
+   JWT = NextAuthJWT(
+       secret=os.getenv("NEXTAUTH_SECRET"),
+   )
+   ```
 
-You should also set the `NEXTAUTH_URL` environment variable, as it is used to determine
-whether secure cookies are being used. Or you can set the cookie names manually.
+2. **HTTPS in Production**: Ensure `NEXTAUTH_URL` starts with `https://` in production to enable secure cookies
 
-## NextAuth / Auth.js v4 compatibility
-A compatibility shim is included if you're still on NextAuth v4, just replace `NexAuthJWT` with `NextAuthJWTv4`
+3. **CSRF Protection**: Keep CSRF protection enabled in production environments
 
 ## Examples
-A [simple example](https://github.com/TCatshoek/fastapi-nextauth-jwt/tree/main/examples/simple) is available in the examples folder. It uses Next.js URL rewrites to direct
-requests to FastAPI. This is just one way to do it, putting both the backend and frontend
-behind something like nginx would also be a good strategy. As long as the cookies can make it to FastAPI
-you should be good to go!
\ No newline at end of file
+
+A [simple example](https://github.com/TCatshoek/fastapi-nextauth-jwt/tree/main/examples/simple) is available in the examples folder. It demonstrates:
+- Using Next.js URL rewrites to route requests to FastAPI
+- Basic JWT validation setup
+- CSRF protection configuration
+
+You can also place both the backend and frontend behind a reverse proxy like nginx, as long as cookies reach FastAPI.
+
+## Environment Variables
+
+- `NEXTAUTH_SECRET`: The secret key used for JWT operations (required)
+- `NEXTAUTH_URL`: The URL of your application (affects secure cookie settings)
+- `ENV`: Set to `"dev"` to disable CSRF protection in development
+
+## Contributing
+
+Contributions are welcome! Please feel free to submit a Pull Request.
+
```

Comment:
I'm very short on time lately, so I can't promise I have time to review PRs in a timely manner :(